lib/ansible/modules/database/mongodb/mongodb_info.py

```diff
@@ -0,0 +1,444 @@
+      and whether it will be validated if provided.
+    required: no
+    type: str
+    default: 'CERT_REQUIRED'
+    choices: ['CERT_NONE', 'CERT_OPTIONAL', 'CERT_REQUIRED']
+
+notes:
+    - Requires the pymongo Python package on the remote host, version 2.4.2+.
+
+requirements: [ 'pymongo' ]
+'''
+
+EXAMPLES = r'''
+- name: Gather all supported information
+  mongodb_info:
+    login_user: admin
+    login_password: secret
+  register: result
+
+- name: Show gathered info
+  debug:
+    msg: '{{ result }}'
+
+- name: Gather only information about databases and their total size
+  mongodb_info:
+    login_user: admin
+    login_password: secret
+    filter: databases, total_size
+
+- name: Gather all information except parameters
+  mongodb_info:
+    login_user: admin
+    login_password: secret
+    filter: '!parameters'
+'''
+
+RETURN = r'''
+general:
+  description: General instance information.
+  returned: always
+  type: dict
+  sample: {"allocator": "tcmalloc", "bits": 64, "storageEngines": ["biggie"], "version": "4.2.3", "maxBsonObjectSize": 16777216}
+databases:
+  description: Database information.
+  returned: always
+  type: dict
+  sample: {"admin": {"empty": false, "sizeOnDisk": 245760}, "config": {"empty": false, "sizeOnDisk": 110592}}
+total_size:
+  description: Total size of all databases in bytes.
+  returned: always
+  type: int
+  sample: 397312
+users:
+  description: User information.
+  returned: always
+  type: dict
+  sample: {"new_user": {"_id": "config.new_user", "db": "config", "mechanisms": ["SCRAM-SHA-1", "SCRAM-SHA-256"], "roles": []}}
+roles:
+  description: Role information.
+  returned: always
+  type: dict
+  sample: {"restore": {"db": "admin", "inheritedRoles": [], "isBuiltin": true, "roles": []}}
+parameters:
+  description: Server parameters information.
+  returned: always
+  type: dict
+  sample: {"maxOplogTruncationPointsAfterStartup": 100, "maxOplogTruncationPointsDuringStartup": 100, "maxSessions": 1000000}
+'''
+
+import traceback
+
+from uuid import UUID
+
+import ssl as ssl_lib
+from distutils.version import LooseVersion
+
+PYMONGO_IMP_ERR = None
+try:
+    from pymongo import version as PyMongoVersion
+    from pymongo import MongoClient
+except ImportError:
+    PYMONGO_IMP_ERR = traceback.format_exc()
+    pymongo_found = False
+else:
+    pymongo_found = True
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import iteritems
+
+
+# =========================================
+# MongoDB module specific support methods.
+#
+
+def check_compatibility(module, srv_version, driver_version):
+    """Check the compatibility between the driver and the database.
+
+    See: https://docs.mongodb.com/ecosystem/drivers/driver-compatibility-reference/#python-driver-compatibility
+
+    Args:
+        module: Ansible module.
+        srv_version (str): MongoDB server version.
```

Comment:
Shouldn't these be of type `LooseVersion`? Or does comparison between `str` and `LooseVersion` work? (If yes, is there a minimal version of `distutils` needed for this?)

Reply:
thanks for pointing this out, changed